src/libclang/preprocessor.cpp

```diff
@@ -1024,7 +1024,8 @@ ts::optional<linemarker> parse_linemarker(position& p)
 
     std::string file_name;
     for (; !starts_with(p, "\""); p.skip())
-        file_name += *p.ptr();
+        if (!(*p.ptr() == '\\' && !file_name.empty() && file_name.back() == '\\'))
```

Comment:
Okay, I'm nitpicking here, but: the logic here is basically "if the current character is a backslash, add two instead of just ones. So can you write it as:

```cpp
if (*p.ptr() == '\\')
    file_name += "\\\\"; // two backslashes
else
    file_name += *p.ptr();
```